src/modes/edit_message.cc

```diff
@@ -134,27 +135,41 @@ namespace Astroid {
     id = edit_id++;
 
     ustring _mid = "";
-    msg_time = time(0);
+
+    /* defaults */
+    accounts = astroid->accounts;
 
 # ifndef DISABLE_PLUGINS
     if (!astroid->plugin_manager->astroid_extension->generate_mid (_mid)) {
 # endif
 
-      char _hostname[1024];
-      _hostname[1023] = 0;
-      gethostname (_hostname, 1023);
-
-      char _domainname[1024];
-      _domainname[1023] = 0;
-      if (getdomainname (_domainname, 1023) < 0) {
-        *_domainname = '\0';
-      }
-
+      //
+      // generate left-hand side of the message id
+      //
+      // get value for right-hand side from config file
       ustring hostname = astroid->config ().get <string> ("mail.message_id_fqdn");
       UstringUtils::trim (hostname);
+      // if hostname contains a "%1", insert the FQDN of the default
+      // account's email address
+      if (hostname.find ("%1", 0) != std::string::npos) {
```

Comment:
If there are no %1 the compose below will just silently not do any replacements, I believe you don't need to have this check then?

Reply:
You're right, in principle the check is not needed. By having the `if` whe can avoid having to access the default account  and having boost strings perform its little dance each and every time.

No strong feelings about this one.

Reply:
Ok, I don't see any immediate problem. The only issue is if there is going to be a `%2` in the future.